packages/cli/src/PublicApi/v1/handlers/credentials/credentials.service.ts

```diff
@@ -53,18 +52,16 @@ export async function saveCredential(
 	encryptedData: ICredentialsDb,
 ): Promise<CredentialsEntity> {
 	await Container.get(ExternalHooks).run('credentials.create', [encryptedData]);
-	void Container.get(InternalHooks).onUserCreatedCredentials({
-		user,
-		credential_name: credential.name,
-		credential_type: credential.type,
-		credential_id: credential.id,
-		public_api: true,
-	});
+	const project = await Container.get(SharedCredentialsRepository).findCredentialOwningProject(
+		credential.id,
+	);
```

Comment:
This will return `undefined` because the credential is only persisted on line 68.

We could move the event emitting code after the transaction, that way everything would be in place in the db and additionally we wouldn't emit the event for credentials that failed to save:

```ts
export async function saveCredential(
	credential: CredentialsEntity,
	user: User,
	encryptedData: ICredentialsDb,
): Promise<CredentialsEntity> {
	const result = await Db.transaction(async (transactionManager) => {
		const savedCredential = await transactionManager.save<CredentialsEntity>(credential);

		savedCredential.data = credential.data;

		const newSharedCredential = new SharedCredentials();

		const personalProject = await Container.get(ProjectRepository).getPersonalProjectForUserOrFail(
			user.id,
			transactionManager,
		);

		Object.assign(newSharedCredential, {
			role: 'credential:owner',
			credentials: savedCredential,
			projectId: personalProject.id,
		});

		await transactionManager.save<SharedCredentials>(newSharedCredential);

		return savedCredential;
	});

	await Container.get(ExternalHooks).run('credentials.create', [encryptedData]);

	const project = await Container.get(SharedCredentialsRepository).findCredentialOwningProject(
		credential.id,
	);
	Container.get(EventService).emit('credentials-created', {
		user,
		credentialType: credential.type,
		credentialId: credential.id,
		projectId: project?.id,
		projectType: project?.type,
		publicApi: true,
	});

	return result;
}
```